be/src/formats/orc/apache-orc/c++/src/Reader.cc

```diff
@@ -1016,7 +1016,8 @@ void RowReaderImpl::buildIORanges(std::vector<InputStream::IORange>* io_ranges)
     for (const proto::Stream& stream : currentStripeFooter.streams()) {
         uint32_t columnId = stream.column();
         uint64_t length = stream.length();
-        if (selectedColumns[columnId] || lazyLoadColumns[columnId]) {
+        // ColumnId = 0 is root column, we always need it
+        if (columnId == 0 || selectedColumns[columnId] || lazyLoadColumns[columnId]) {
             io_ranges->emplace_back(InputStream::IORange{.offset = offset, .size = length});
```

Comment:
is this a bug?

Reply:
<img width="1154" alt="image" src="https://github.com/StarRocks/starrocks/assets/18729228/15ba5a2a-ae55-4a5d-bc1b-1ef5605fc3eb">

ColumnId = 0 is root node, must always include it.